src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -312,34 +317,71 @@ private function getAccessorMethod($class, $property)
     private function getMutatorMethod($class, $property)
     {
         $ucProperty = ucfirst($property);
+        $names = $this->getSingulars($ucProperty);
 
         foreach (self::$mutatorPrefixes as $prefix) {
-            try {
-                $reflectionMethod = new \ReflectionMethod($class, $prefix.$ucProperty);
+            if (in_array($prefix, self::$arrayMutatorPrefixes)) {
+                array_unshift($names, $ucProperty);
```

Comment:
Shouldn't this happen outside of the loop? We seem to be prepending repeatedly... (And I don't understand the need for this check.)